http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/NonLeoMessageSubscriber.scala

```diff
@@ -123,37 +129,47 @@ class NonLeoMessageSubscriber[F[_]](gkeAlg: GKEAlgebra[F],
           // We mark the runtime as Deleted, and delete the instance.
           // If instance deletion fails for some reason, it will be cleaned up by resource-validator
           _ <- runtimeFromGoogle.traverse { instance =>
-            for {
-              runtimeInfo <- clusterQuery
-                .getActiveClusterRecordByName(msg.googleProject, RuntimeName(instance.getName))
-                .transaction
-              _ <- runtimeInfo.traverse { runtime =>
-                for {
-                  _ <- clusterQuery
-                    .markDeleted(msg.googleProject, RuntimeName(instance.getName), ctx.now, Some("cryptomining"))
-                    .transaction
-                  _ <- computeService.deleteInstance(msg.googleProject,
-                                                     msg.resource.labels.zone,
-                                                     InstanceName(msg.resource.labels.instanceId.toString))
-                  userSubjectId <- samDao.getUserSubjectId(runtime.auditInfo.creator, runtime.googleProject)
-                  _ <- userSubjectId.traverse { sid =>
-                    val byteString = ByteString.copyFromUtf8(CryptominingUserMessage(sid).asJson.noSpaces)
-
-                    val message = PubsubMessage
-                      .newBuilder()
-                      .setData(byteString)
-                      .putAttributes("traceId", ctx.traceId.asString)
-                      .putAttributes("cryptomining", "true")
-                      .build()
-                    publisher.publishNativeOne(message)
-                  }
-                } yield ()
-              }
-            } yield ()
+            deleteCryptominingRuntime(msg.googleProject,
+                                      RuntimeName(instance.getName),
+                                      msg.resource.labels.zone,
+                                      "custom detector")
           }
         } yield ()
     } yield ()
 
+  private[monitor] def deleteCryptominingRuntime(
+    googleProject: GoogleProject,
+    runtimeName: RuntimeName,
+    zone: ZoneName,
+    message: String
+  )(implicit ev: Ask[F, AppContext]): F[Unit] =
+    for {
+      ctx <- ev.ask
+      runtimeInfo <- clusterQuery
+        .getActiveClusterRecordByName(googleProject, runtimeName)
+        .transaction
+      _ <- runtimeInfo.traverse { runtime =>
```

Comment:
Should we maybe log something or send a metric if we can't find the runtime in the Leo DB? Just thinking the SCC alerts may potentially pick up non-Leo VMs.

Reply:
oh, that's a good point...I'll add a log message for now..think we can extract metrics from log message as well